Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/PostgresCentralRepo.java

```diff
@@ -198,11 +198,11 @@ protected Connection connect() throws CentralRepoException {
             if (connectionPool == null) {
                 setupConnectionPool();
             }
-        }
-        try {
-            return connectionPool.getConnection();
-        } catch (SQLException ex) {
-            throw new CentralRepoException("Error getting connection from connection pool.", Bundle.PostgresEamDb_connectionFailed_message(), ex); // NON-NLS
+            try {
+                return connectionPool.getConnection();
+            } catch (SQLException ex) {
+                throw new CentralRepoException("Error getting connection from connection pool.", Bundle.PostgresEamDb_connectionFailed_message(), ex); // NON-NLS
+            }
```

Comment:
I'd be really, really surprised if the connection pool getConnection() method is not thread-safe.

Reply:
Not getConnection(), but the interaction with the `connectionPool` variable is not thread-safe

Reply:
Yes, of course. You are correct.